src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,169 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String INDEX_LESS_THAN_ZERO_MESSAGE = "Index can't be less than zero";
+    private static final String INDEX_NOT_FOUND_MESSAGE = "Not found index: ";
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            insertFirst(value);
+            return;
+        }
+        Node<T> newNode = new Node<>(tail, value, null);
+        tail.next = newNode;
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (size == index) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        if (index == 0) {
+            insertFirst(value);
+            return;
+        }
+        Node<T> prevNode = elementAt(index - 1);
```

Comment:
why index - 1?

Reply:
Because prevNode is the node with the previous index

Reply:
Oh, I understood